tests/system/lab/test

```diff
@@ -604,32 +604,147 @@ event_toggled_sent_clear () {
+        if [[ -n "$EVENT_NAME" ]] && [[ "$EVENT_NAME" == "\"test\"" ]] && \
+        [[ -n "$RULE" ]] && [[ "$RULE" == "\"Threshold\"" ]] && \
+        [[ -n "$TYPE" ]] && [[ "$TYPE" == "\"toggled\"" ]]; then break; fi
+    done
+
+    if [[ ${LOOP} -eq ${MAX_RETRIES} ]];
+      then
+        display_and_collect_err "TIMEOUT! event instance not created successfully . $URL/notification";
+        if [[ ${EXIT_EARLY} -eq 1 ]]; then exit 1; fi
+      else
+        echo "---- event instance created successfully ----"
+    fi
+
     echo "Set rule"
     SET_RULE=`curl -sX PUT "$URL/category/ruletest" -d '{"asset":"sinusoid", "datapoint":"sinusoid", "trigger_value": "0.8"}' | jq .`
     echo "$SET_RULE"
 
+    for LOOP in $(seq ${MAX_RETRIES}); do
+      RESULT=`curl -sX GET "$URL/category/ruletest"`
+      echo ${RESULT}
+      ASSET=`echo ${RESULT} | jq .asset.value`
+      DATAPOINT=`echo ${RESULT} | jq .datapoint.value`
+      TRIGGER_VALUE=` echo ${RESULT} | jq .trigger_value.value`
+      if [[ -n "$ASSET" ]] && [[ "$ASSET" == "\"sinusoid\"" ]] &&\
+      [[ -n "$DATAPOINT" ]] && [[ "$DATAPOINT" == "\"sinusoid\"" ]] &&\
+      [[ -n "$TRIGGER_VALUE" ]] && [[ "$TRIGGER_VALUE" == "\"0.8\"" ]]; then break; fi
+    done
+
+    if [[ ${LOOP} -eq ${MAX_RETRIES} ]];
+      then
+        display_and_collect_err "TIMEOUT! Rule not  set successfully . $URL/category/ruletest";
+        if [[ ${EXIT_EARLY} -eq 1 ]]; then exit 1; fi
+      else
+        echo "---- Rule set successfully ----"
+    fi
+
     echo "Set delivery"
     SET_DELIVERY_CHN=`curl -sX PUT "$URL/category/deliverytest" -d '{"asset": "sin0.8", "description":"asset notification", "enable":"true"}' | jq .`
     echo "$SET_DELIVERY_CHN"
 
-    # TODO: exact time to sleep?
-    echo "Sleeping for 20 seconds"
+    for LOOP in $(seq ${MAX_RETRIES}); do
+      RESULT=`curl -sX GET "$URL/category/deliverytest"`
+      echo ${RESULT}
+      ASSET=`echo ${RESULT} | jq .asset.value`
+      ENABLE_VALUE=`echo ${RESULT} | jq .enable.value`
+      if [[ -n "$ASSET" ]] && [[ "$ASSET" == "\"sin0.8\"" ]] &&\
+      [[ -n "$ENABLE_VALUE" ]] && [[ "$ENABLE_VALUE" == "\"true\"" ]]; then break; fi
+    done
+
+    if [[ ${LOOP} -eq ${MAX_RETRIES} ]];
+      then
+        display_and_collect_err "TIMEOUT! Delivery channel not set successfully . $URL/category/deliverytest";
+        if [[ ${EXIT_EARLY} -eq 1 ]]; then exit 1; fi
+      else
+        echo "---- Delivery channel set successfully ----"
+    fi
+
+    echo "Sleeping for 20 seconds until sin0.8 is added in asset table"
     sleep 20
 
-    # TODO: verify asset tracker has entry for sin0.8
+    for LOOP in $(seq ${MAX_RETRIES}); do
+        RESULT=`curl -sX GET "$URL/track"`
+        echo ${RESULT}
+        ASSET_NAME=`echo ${RESULT} | jq '.track[]|select(.asset=="sin0.8").asset'`
+        if [[ -n "$ASSET_NAME" ]] && [[ "$ASSET_NAME" == "\"sin0.8\"" ]]; then break; fi
+    done
```

Comment:
The asset created by asset notification delivery plugin does not create an entry in asset tracker table. this is bug! Please check and log if not exists.

Also We should verify `ASSET_CREATION_EVENT` event and service created it i.e. `ASSET_CREATED_BY` .